### PR TITLE
adding monetization lables to API image

### PR DESCRIPTION
**Pull request by CrowleyRajapakse**
## Issue
product-apim: https://github.com/wso2/product-apim/issues/6760
product-apim: https://github.com/wso2/product-apim/issues/6699

## Methodology

- Changing the APIInfo REST API to send "monetizationLabel" which can be "FREE", "PAID" ,"FREEMIUM" and the logic to check which label for the API.
- Changing the settings rest api to get the monetization enabled value from tenant.conf.
- Retrieving the settings response using settings context in store and passing the enableMonetization attribute to API Context.
- Checking if the monetization enabled and adding the label to the API Image Card.

- Adding number validation for API Key timeout field using joi validation library. Now the generate button is active only when numbers are provided and error msg is displayed as helper text.

## Labels
APIM 3.0.1 UI REST WUM

## Images 

Screen Shot 2019-11-07 at 11 35 40 AM

Screen Shot 2019-11-08 at 3 05 49 PM

### CODE REVIEW

**Comment by CrowleyRajapakse** [comment]
Screen Shot 2019-11-07 at 2 41 24 PM

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Listing/ApiThumb.jsx [hunk 1]
```diff
@@ -292,6 +302,9 @@ class ApiThumb extends React.Component {
                 raised={isHover}
                 className={classNames('image-thumbnail', classes.card)}
             >
+                {this.context.isMonetizationEnabled && (
```
Comment (dushansilva) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Listing/ApiThumb.jsx]: shall we use de-structuring here?

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Listing/CommonListing.jsx [hunk 2]
```diff
@@ -308,12 +323,12 @@ class CommonListing extends React.Component {
                     {active && allTags && allTags.length > 0 && <ApiBreadcrumbs selectedTag={selectedTag} />}
                     <div className={classes.listContentWrapper}>
                         {listType === 'grid' && (
-                            <ApiContext.Provider value={{ apiType }}>
+                            <ApiContext.Provider value={{ apiType, isMonetizationEnabled }}>
```
Comment (dushansilva) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Listing/CommonListing.jsx]: Why are we creating a new context here again?

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Shared/ApiKey.jsx [hunk 3]
```diff
@@ -61,7 +61,11 @@ const tokens = (props) => {
                 }
                 break;
             case 'timeout':
-                newRequest.timeout = currentTarget.value;
+                if (!isNaN(currentTarget.value)) {
```
Comment (dushansilva) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Shared/ApiKey.jsx]: is this a validation? if so do we show an error when it is invalid ?

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Listing/CommonListing.jsx (hunk 2 above)
Comment (CrowleyRajapakse) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Listing/CommonListing.jsx]: API context was already created previously, i only needed to add monetization enabled value to the API context. Now in the API Context API type and isMonetizationEnabled attributes are available.

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/APIMappingUtil.java [hunk 4]
```diff
@@ -618,6 +618,21 @@ static APIInfoDTO fromAPIToInfoDTO(API api) {
         String subscriptionAllowedTenants = api.getSubscriptionAvailableTenants();
         apiInfoDTO.setIsSubscriptionAvailable(isSubscriptionAvailable(apiTenant, subscriptionAvailability,
                 subscriptionAllowedTenants));
+        int free = 0, commercial = 0;
+        for (Tier tier : throttlingPolicies) {
+            if(tier.getTierPlan().equalsIgnoreCase("FREE")) {
+                free = free + 1;
+            } else if (tier.getTierPlan().equalsIgnoreCase("COMMERCIAL")) {
```
Comment (dushansilva) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/APIMappingUtil.java]: Shall we use constants for these?

Comment (CrowleyRajapakse) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/mappings/APIMappingUtil.java]: resolved

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Shared/ApiKey.jsx (hunk 3 above)
Comment (CrowleyRajapakse) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Shared/ApiKey.jsx]: removed the Nan validation and used the joi library to validate

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Listing/ApiThumb.jsx (hunk 1 above)
Comment (CrowleyRajapakse) [reply features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Listing/ApiThumb.jsx]: resolved

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/SettingsApiServiceImpl.java [hunk 5]
```diff
@@ -49,12 +51,15 @@
...
+            String tenantDomain = MultitenantUtils.getTenantDomain(username);
+            APIConsumer apiConsumer = RestApiUtil.getConsumer(username);
+            boolean monetizationEnabled = apiConsumer.isMonetizationEnabled(tenantDomain);
             boolean isUserAvailable = false;
             if (!APIConstants.WSO2_ANONYMOUS_USER.equalsIgnoreCase(username)) {
                 isUserAvailable = true;
             }
             SettingsMappingUtil settingsMappingUtil = new SettingsMappingUtil();
-            SettingsDTO settingsDTO = settingsMappingUtil.fromSettingstoDTO(isUserAvailable);
+            SettingsDTO settingsDTO = settingsMappingUtil.fromSettingstoDTO( isUserAvailable, monetizationEnabled );
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/SettingsApiServiceImpl.java]: formatting issue

**Pull request merged**